modules/service/src/main/scala/ch/epfl/bluebrain/nexus/kg/service/routes/DefaultRouteHandling.scala

```diff
@@ -37,4 +55,32 @@ abstract class DefaultRouteHandling()(implicit prefixes: PrefixUris) {
   private def combine(cred: Option[OAuth2BearerToken]) =
     writeRoutes(cred) ~ readRoutes(cred) ~ searchRoutes(cred)
 
+  /**
+    * Serializes and formats an entity according to the given JSON-LD format via implicitly available instances
+    * of Encoder and ToEntityMarshaller.
+    *
+    * @param entity the entity to be serialized
+    * @param format the JSON-LD output format
+    * @tparam A a generic type parameter for the entity
+    * @return an Akka HTTP Route
+    */
+  protected def formatOutput[A](entity: A, format: JsonLdFormat)(implicit encoder: Encoder[A],
+                                                                 marshaller: ToEntityMarshaller[A],
+                                                                 keys: OrderedKeys): Route = {
+    format match {
+      case JsonLdFormat.Default => complete(StatusCodes.OK -> entity)
+      case f: JenaJsonLdFormat =>
+        onSuccess(expander(encoder(entity), f.toJena)) { formatted =>
+          complete(StatusCodes.OK -> formatted)
```

Comment:
I think the content-type will be set to `text/plain` instead of `application/ld+json` for this case.

Reply:
Can @hygt verify that on the tests?